lib/al/Library/Area/AreaObjGroup.cpp

```diff
@@ -0,0 +1,64 @@
+#include "al/Library/Area/AreaObjGroup.h"
+
+#include <basis/seadNew.h>
+
+namespace al {
+AreaObjGroup::AreaObjGroup(const char* groupName, const int capacity) {
+    mGroupName = groupName;
+    mBuffer = nullptr;
+    mCount = 0;
+    mCapacity = capacity;
+}
+
+void AreaObjGroup::createBuffer() {
+    if (mCapacity < 1) {
+        return;
+    }
+    mBuffer = new AreaObj*[mCapacity];
+}
+
+void AreaObjGroup::createBuffer(s32 capacity) {
+    mCapacity = capacity;
+    if (mCapacity < 1) {
+        return;
+    }
+    mBuffer = new AreaObj*[mCapacity];
+}
+
+AreaObj* AreaObjGroup::getAreaObj(const s32 index) const {
+    return mBuffer[index];
+}
+
+AreaObj* AreaObjGroup::getInVolumeAreaObj(const sead::Vector3f& position) const {
+    al::AreaObj *areaObj, *toReturn, *curAreaObj;
```

Comment:
Do not declare like this, but specify the types at the place where they really are initialized/declared